### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by shaaibu7**
- [ ] Create a new Rails app called 'hello-rails-react'.
- [ ] Verify that the bundling-rails gem is added to your Gemfile.
- [ ] Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- [ ] Update webpack.config.js to include the babel loader.
- [ ] Make sure that your project has a Postgres database set up.
- [ ] Create a table for storing your messages and create 5 different greetings.
- [ ] Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- [ ] Create a static view that will be the root of your app.
- [ ] Create the App component with react-router and render it in your static view.
- [ ] Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- [ ] Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- [ ] Display the random greeting in your Greeting component.

### CODE REVIEW

File: app/javascript/components/Greetings.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+const Greetings = () => {
+  const dispatch = useDispatch();
+  const { isLoading, greeting } = useSelector((state) => state.greetings.greeting);
+  useEffect(() => {
+    dispatch(fetchGreeting());
+  }, []);
+  if (isLoading) {
+    return <h1>Loading...</h1>;
+  }
```
Comment (elisha2kyakpo1) [inline app/javascript/components/Greetings.js:13]: - [ ] Kindly fix the display of messages on the screen. There is an error on the browser console, maybe it's want is causing the app to show nothing on the screen.

hello-rails

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @shaaibu7, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
👏 The seed file was loaded with data.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @shaaibu7,

✔️ Creation of Greeting component

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bienvenuushindi — APPROVED** [review]
# STATUS: APPROVED 🟢 

## Hi @shaaibu7   👋, 

Well done implementing the requested changes 👏
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Highlights 🎯 
:+1: No linter errors✔️
:+1: Good PR title
:+1: Descriptive PR Summary

### Optional Suggestions



**_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**